### PR TITLE
Minor random fixes found on aarch64

### PR DESCRIPTION
**Pull request by superm1**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/mtd/mtd.quirk [hunk 1]
```diff
@@ -16,3 +16,21 @@ Flags = no-probe
 [MTD\NAME_BIOS]
 Name = Internal SPI Controller
 #FirmwareGType = FuIfdFirmware
+
+# AMD Kria System on Module
+[MTD\VENDOR_xlnx&NAME_Image-A-(FSBL,-PMU,-ATF,-U-Boot)]
```
[the remark this answers is not in the document]
Comment (superm1) [reply plugins/mtd/mtd.quirk:21]: Couldn't that break instance Id for existing devices? If you don't think there are any in practice then yeah I can clean it up.

[the remark this answers is not in the document]
Comment (hughsie) [reply plugins/mtd/mtd.quirk:21]: The only ones I know of are the pinephone ones.

File: libfwupdplugin/fu-common.c [hunk 2]
```diff
@@ -305,14 +305,17 @@ fu_error_convert(GError **perror)
...
 
 	/* sanity check */
 	if (error == NULL)
 		return;
 
 	/* find match */
-	for (guint i = 0; i < G_N_ELEMENTS(map); i++) {
+	for (guint i = 0;; i++) {
+		if (map[i].domain == FWUPD_ERROR && map[i].fwupd_code == FWUPD_ERROR_LAST)
+			break;
```
Comment (hughsie) [inline libfwupdplugin/fu-common.c:318]: can you just wrap this in the #ifdef too? Or drop this commit and review https://github.com/fwupd/fwupd/pull/7412 instead as it deletes the entire section :)

Comment (superm1) [reply libfwupdplugin/fu-common.c]: Dropped change. Will get into that review later on, thx!

File: plugins/mtd/mtd.quirk (hunk 1 above)
[the remark this answers is not in the document]
Comment (superm1) [reply plugins/mtd/mtd.quirk:21]: Oh but I'll need to adjust the quirk now too

File: plugins/mtd/mtd.quirk [hunk 3]
```diff
@@ -16,3 +16,21 @@ Flags = no-probe
 [MTD\NAME_BIOS]
 Name = Internal SPI Controller
 #FirmwareGType = FuIfdFirmware
+
+# AMD Kria System on Module
+[MTD\VENDOR_xlnx&PRODUCT_ZynqMP-KV260-revB&NAME_Image-A-FSBL,-PMU,-ATF,-U-Boot]
```
Comment (hughsie) [inline plugins/mtd/mtd.quirk:21]: I think we want comma in the strsafe list too -- we use that in some places to split the instance-ids.

**Review by hughsie — APPROVED** [review]

**Pull request merged**